### PR TITLE
Add get_config to the ethernet API

### PR DESCRIPTION
**Pull request by tgorochowik**
This PR extends the ethernet_api with a get_config callback and extends the ethernet management API to make use of it. For now it only supports getting the number of priority queues.

Note that this will be used for a proper Qav implementation for SAM GMAC once support for multiple HW (priority) queues is merged (part of #8622)

Please review, thanks!

### CODE REVIEW

File: subsys/net/l2/ethernet/ethernet_mgmt.c [hunk 1]
```diff
@@ -128,6 +128,45 @@ static int ethernet_set_config(u32_t mgmt_request,
 	return api->set_config(net_if_get_device(iface), type, &config);
 }
 
+static int ethernet_get_config(u32_t mgmt_request,
```
Comment (tbursztyka) [inline subsys/net/l2/ethernet/ethernet_mgmt.c]: move that function before line 191
The untold rule here is: 
- first the function
- then the relevant NET_MGMT_REGISTER_REQUEST_HANDLER() macro call(s) relevant to that function

File: tests/net/ethernet_mgmt/src/main.c [hunk 2]
```diff
@@ -109,8 +111,9 @@ static int eth_fake_set_config(struct device *dev,
 		break;
 	case ETHERNET_CONFIG_TYPE_QAV_DELTA_BANDWIDTH:
 	case ETHERNET_CONFIG_TYPE_QAV_IDLE_SLOPE:
-		/* Assumes just one priority queue - validate the id*/
-		if (config->qav_queue_param.queue_id != 0) {
+		queue_id = config->qav_queue_param.queue_id;
+
```
Comment (tbursztyka) [inline tests/net/ethernet_mgmt/src/main.c]: you can get rid of that empty line as your if is testing the previous assignment

**Review by tbursztyka — CHANGES_REQUESTED** [review]
very small comments

File: subsys/net/l2/ethernet/ethernet_mgmt.c (hunk 1 above)
Comment (tgorochowik) [reply subsys/net/l2/ethernet/ethernet_mgmt.c]: Sure! That makes perfect sense

File: tests/net/ethernet_mgmt/src/main.c (hunk 2 above)
Comment (tgorochowik) [reply tests/net/ethernet_mgmt/src/main.c]: That is right, thanks!

**Comment by tgorochowik** [comment]
@tbursztyka Thanks!

Updated according to review.

File: include/net/ethernet.h [hunk 3]
```diff
@@ -81,6 +81,9 @@ enum ethernet_hw_caps {
 
 	/** Promiscuous mode supported */
 	ETHERNET_PROMISC_MODE		= BIT(10),
+
+	/** Priority queues available */
+	ETHERNET_PRIORITY_QUEUES	= BIT(11),
```
Comment (jukkar) [inline include/net/ethernet.h]: Just a note that this will conflict with #9153 PR as it is adding a bit here too.

**Review by jukkar — APPROVED** [review]
LGTM

**Review by tbursztyka — APPROVED** [review]

**Pull request merged**